rustls-libssl/src/entry.rs

```diff
@@ -470,6 +470,16 @@ entry! {
     }
 }
 
+pub type SSL_CTX_cert_cb_func =
+    Option<unsafe extern "C" fn(ssl: *mut SSL, arg: *mut c_void) -> c_int>;
+
+entry! {
+    pub fn _SSL_CTX_set_cert_cb(ctx: *mut SSL_CTX, cb: SSL_CTX_cert_cb_func, arg: *mut c_void) {
+        let ctx = try_clone_arc!(ctx);
+        ctx.get_mut().set_cert_cb(cb, arg);
```

Comment:
Should this be checking that `cb` isn't `NULL`?

Reply:
the `SSL_CTX_cert_cb_func` type is `Option<unsafe extern "C" fn(ssl: *mut SSL, arg: *mut c_void) -> c_int>;` -- in this case it's relying on the NULL niche in function pointers to encode `NULL` as `Option::None`, which is nice as it forces every user of the callbacks to consider their nullability in a more rusty way.

the other answer to this is: yes, `SSL_CTX_set_cert_cb(ctx, NULL, ...)` should be allowed as a method to clear any previously set callback.

Reply:
> in this case it's relying on the NULL niche in function pointers to encode NULL as Option::None

Aha! I think this was the aspect I was overlooking. Thanks :+1: